macros/src/codegen.rs

```diff
@@ -596,13 +630,20 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
                 mut event: #events_type_name <#event_lifetimes>
             ) -> Result<&#states_type_name <#state_lifetimes>, #error_type> {
                 self.context.log_process_event(self.state()?, &event);
+
                 match self.state.take().ok_or_else(|| #error_type_name ::Poisoned)? {
                     #(#states_type_name::#in_states => match event {
                         #(#events_type_name::#events => {
                             #code_blocks
-
-                            self.state()
+                            #[allow(unreachable_code)]
+                            {
+                                // none of the guarded or non-guarded transitions occurred,
+                                // therefore return an error,
+                                self.state = Some(#states_type_name::#in_states);
```

Comment:
Why are we modifying the internal state if none of the transitions happened?

```suggestion
```

Reply:
removed